File: manual/src/test/java/ActorDocTest.java
import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

import org.junit.jupiter.api.Test;
import org.shakespeareframework.Actor;

class ActorDocTest {

  @Test
  void act1() {
    // tag::create-actor[]
    var robin = new Actor("Robin");
    // end::create-actor[]
    assertThat(robin.getName()).isEqualTo("Robin");
  }

  @Test
  void act2() {
    // tag::create-actor-random-name[]
    var user = new Actor();
    // end::create-actor-random-name[]
    assertThat(user.getName()).isNotBlank();
  }
}

File: manual/src/test/kotlin/examples/kotlin/ActorDocTest.kt
package examples.kotlin

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.shakespeareframework.Actor

class ActorDocTest {

  @Test
  fun act1() {
    // tag::create-actor[]
    val robin = Actor("Robin")
    // end::create-actor[]
    assertThat(robin.name).isEqualTo("Robin")
  }

  @Test
  fun act2() {
    // tag::create-actor-random-name[]
    val user = Actor()
    // end::create-actor-random-name[]
    assertThat(user.name).isNotBlank()
  }
}
